config/7.1.0/securebanking/ig/scripts/groovy/com/forgerock/sapi/gateway/oauth/OAuthErrorResponseFactory.groovy

```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    String getHtmlErrorMessage(Form errorForm) {
+        logger.debug("{}getHtmlErrorMessage, errorForm: '{}'", logPrefix,  errorForm)
+        StringBuilder errorMessageBuilder = new StringBuilder("<!doctype html><html lang=\"en\"><head><meta charset=\"utf-8\"><title>Authorization Error" + 
+          "</title><meta name=\"viewport\" content=\"width=device-width, initial-scale=1\"></head><body>")
+        errorForm.forEach( (key, val) -> {
+            logger.debug("{}processing entry for {}, val is {}", logPrefix, key, val)
+            errorMessageBuilder.append("<p>" + key + ": ")
+            List<String> values = val
+            Boolean first = true
+            values.forEach( (innerVal) -> {
+                logger.debug("{}adding value {} to error message", logPrefix, innerVal)
+                if(!first){
+                    errorMessageBuilder.append(', ')
+                    first = false
+                }
+                errorMessageBuilder.append(innerVal)
+            })
+            errorMessageBuilder.append('</p>')
+        } )
+        errorMessageBuilder.append("</body></html>")
+        String htmlErrorMessage =  errorMessageBuilder.toString()
+        logger.debug("{}html error message is {}", logPrefix, htmlErrorMessage)
+        return htmlErrorMessage
+    }
+
+    String getJsonErrorMessage(Form errorForm){
+        logger.debug("{}getJsonErrorMessage, errorForm: '{}'", logPrefix, errorForm)
+        String jsonErrorMessage = new String(Json.writeJson(errorForm))
+        logger.debug("{}html error message is {}", logPrefix, jsonErrorMessage)
+        return jsonErrorMessage
+    }
+
+    String getTextErrorMessage(Form errorForm) {
+        logger.debug("{}getTextErrorMessage, errorForm: '{}'", logPrefix, errorForm)
+        StringBuilder errorMessageBuilder = new StringBuilder()
+        errorForm.forEach( (key, val) -> {
+            logger.debug("{}processing entry for {}, val is {}", logPrefix, key, val)
+            errorMessageBuilder.append(key + ": ")
+            List<String> values = val
+            Boolean first = true
+            values.forEach( (innerVal) -> {
+                logger.debug("{}adding value {} to error message", logPrefix, innerVal)
+                if(!first){
+                    errorMessageBuilder.append(', ')
+                    first = false
+                }
+                errorMessageBuilder.append(innerVal)
+            })
+            errorMessageBuilder.append('\n')
+        } )
+        String textErrorMessage =  errorMessageBuilder.toString()
+        logger.debug("{}text error message is {}", logPrefix, textErrorMessage)
+        return textErrorMessage
+    }
+
+    String getFormUrlEncodedErrorMessage(Form errorForm) {
+        logger.debug("{}getFormUrlEncodedErrorMessage, errorForm: '{}'", logPrefix, errorForm)
+        StringBuilder errorMessageBuilder = new StringBuilder()
+        errorForm.forEach( (key, val) -> {
```

Comment:
I think Form has a convenience method for this: org.forgerock.http.protocol.Form#toQueryString